examples/interoperability/app.js

```diff
@@ -0,0 +1,125 @@
+import DeckGL, {
+  ScatterplotLayer, OrthographicViewport, COORDINATE_SYSTEM
+} from '../../src';
+// import DeckGL, {ScatterplotLayer} from 'deck.gl';
+
+const DEGREE_TO_RADIAN = Math.PI / 180;
+const NUM_POINTS = 2000;
+
+class Root extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this._onResize = this._onResize.bind(this);
+    this._update = this._update.bind(this);
+
+    this.state = {
+      width: 0,
+      height: 0,
+      viewMode: 0
+    };
+
+    this.points = Array.from(Array(NUM_POINTS)).map((_, i) => {
+      return {
+        radius: Math.random(),
+        theta: Math.random() * 360
+      };
+    });
+  }
+
+  componentWillMount() {
+    window.addEventListener('resize', this._onResize);
+    this._onResize();
+  }
+
+  componentDidMount() {
+    window.requestAnimationFrame(this._update);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this._onResize);
+  }
+
+  _onResize() {
+    const {innerWidth: width, innerHeight: height} = window;
+    this.setState({width, height});
+  }
+
+  _update() {
+    this.points.forEach(point => {
+      point.theta += Math.sqrt(point.radius);
+    });
+    this.points._lastUpdate = Date.now();
+
+    this.forceUpdate();
+    window.requestAnimationFrame(this._update);
+  }
+
+  _renderSVGPoints() {
+    const {width, height} = this.state;
+    const size = Math.min(width, height) / 2;
+
+    return (
+      <g transform={`translate(${size}, ${size})`}>
+        {this.points && this.points.length && this.points.map((p, i) => (
+          <circle key={i} r={2} fill="#08F"
+            cx={p.radius * Math.cos(p.theta * DEGREE_TO_RADIAN) * size}
+            cy={p.radius * Math.sin(p.theta * DEGREE_TO_RADIAN) * size}/>
+        ))}
+      </g>
+    );
+  }
+
+  _renderScatterplotLayer() {
+    const {width, height} = this.state;
+    const size = Math.min(width, height) / 2;
+
+    return new ScatterplotLayer({
+      id: 'scatterplot-layer',
+      data: this.points,
+      getPosition: p => [
+        p.radius * Math.cos(p.theta * DEGREE_TO_RADIAN) * size,
+        p.radius * Math.sin(p.theta * DEGREE_TO_RADIAN) * size
+      ],
+      getRadius: p => 2,
+      getColor: p => [255, 0, 128, 196],
+      updateTriggers: {
+        getPosition: this.points._lastUpdate
+      },
+      projectionMode: COORDINATE_SYSTEM.IDENTITY,
+      // there's a bug that the radius calculated with project_scale
+      radiusMinPixels: 2
+    });
+  }
+
+  render() {
+    const {width, height, viewMode} = this.state;
+
+    const left = -Math.min(width, height) / 2;
+    const top = -Math.min(width, height) / 2;
+    const glViewport = new OrthographicViewport({width, height, left, top});
+
+    return width && height && <div>
+      {viewMode > 0 && <svg viewBox={`0 0 ${width} ${height}`}>
+        { this._renderSVGPoints() }
+      </svg>}
+      {viewMode !== 1 && <DeckGL width={width} height={height} viewport={glViewport}
```

Comment:
This is quite a bit of logic to put inline in the JSX. Might be more readable to move this to if/switch before the render, use symbolic constants etc.